src/Homie/Boot/BootConfig.cpp

```diff
@@ -335,7 +335,7 @@ void BootConfig::_onDeviceInfoRequest(AsyncWebServerRequest *request) {
   for (IHomieSetting* iSetting : IHomieSetting::settings) {
     JsonObject& jsonSetting = jsonBuffer.createObject();
 
-    if (String(iSetting->getType()) != "unknown") {
+    if (strcmp(iSetting->getType(), "unknown") != 0) {
```

Comment:
Why strcmp?

Reply:
In the core of the lib, we should try avoid using String. String will copy the chars into dynamic memory even for something as simple as this... Using String everywhere can lead to heep fragmentation.